controllers/volume/volume_controller.go

```diff
@@ -643,7 +657,12 @@ func (r *Reconciler) processAttachments(
 			// CNS noop reconciles of an attachment that is already attached so we must delete the existing
 			// attachment and create a new one.
 			if volume.PersistentVolumeClaim.ClaimName == attachment.Spec.VolumeName {
-				volumeStatus = append(volumeStatus, attachmentToVolumeStatus(volume.Name, attachment))
+				volumeStatus := attachmentToVolumeStatus(volume.Name, attachment)
+				volumeStatus.Used = existingManagedVolUsage[volume.Name]
+				if err := updateVolumeStatusWithLimit(ctx, r.reader, *volume.PersistentVolumeClaim, &volumeStatus); err != nil {
```

Comment:
@bryanv I updated this as we discussed. I left the instance storage optimization since it's free with a loop no longer required.